File: capy/src/e2e-cfun/java/dev/capylang/test/DateTimeDifferenceTest.java
package dev.capylang.test;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class DateTimeDifferenceTest {
    @Test
    void shouldRenderForwardDifferenceAsCanonicalIsoDuration() {
        assertThat(DateTimeDifference.forwardDifferenceIso()).isEqualTo("P1DT1H1M1S");
    }

    @Test
    void shouldRenderBackwardDifferenceAsNegatedCanonicalIsoDuration() {
        assertThat(DateTimeDifference.backwardDifferenceIso()).isEqualTo("P-1DT-1H-1M-1S");
    }

    @Test
    void shouldRoundTripDifferenceThroughPlus() {
        assertThat(DateTimeDifference.differenceRoundTrips()).isTrue();
    }
}

File: capy/src/e2e-cfun/js/dev/capylang/test/DateTimeDifference.test.js
const test = require('node:test');
const { assert, generatedModule } = require('./_support');

test('DateTimeDifference', () => {
    const dateTime = generatedModule('dev/capylang/test/DateTimeDifference.js');
    assert.equal(dateTime.forwardDifferenceIso(), 'P1DT1H1M1S');
    assert.equal(dateTime.backwardDifferenceIso(), 'P-1DT-1H-1M-1S');
    assert.equal(dateTime.differenceRoundTrips(), true);
});
